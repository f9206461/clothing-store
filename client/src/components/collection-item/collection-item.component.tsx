import React from "react";
import { connect } from "react-redux";
import { addItem } from "../../redux/cart/cart.actions";

import { 
    CollectionItemCont, 
    ImgCont, 
    CollectionFooterCont,
    NameCont,
    PriceCont,
    CustomButtonCont
} from "./collection-item.styles";

interface Props {
    item: Item;
    addItem: (item: Item) => void;
}

export const CollectionItem = ({ item, addItem }: Props) => {
    const { name, price, imageUrl } = item;
    
    return (
        <CollectionItemCont>
            <ImgCont className="image" imageUrl={imageUrl} />
            <CollectionFooterCont>
                <NameCont>{name}</NameCont>
                <PriceCont>{price}</PriceCont>
            </CollectionFooterCont>
            <CustomButtonCont inverted onClick={() => addItem(item)}>Add to Cart</CustomButtonCont>
        </CollectionItemCont>
    )
};

const mapDispatchToProps = (dispatch: any) => ({
    addItem: (item: Item) => dispatch(addItem(item))
})

export default connect(null, mapDispatchToProps)(CollectionItem);