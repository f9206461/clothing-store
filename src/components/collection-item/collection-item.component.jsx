import React from "react";
import { connect } from "react-redux";
import { addItem } from "../../redux/cart/cart.actions";

import { 
    CollectionItemCont, 
    ImgCont, 
    CollectionFooterCont,
    NameCont,
    PriceCont,
    CustomButtonCont
} from "./collection-item.styles";

const CollectionItem = ({ item, addItem }) => {
    const { name, price, imageUrl } = item;
    
    return (
        <CollectionItemCont>
            <ImgCont className="image"
                style={{
                    backgroundImage:`url(${imageUrl})`
                }}
            />
            <CollectionFooterCont>
                <NameCont>{name}</NameCont>
                <PriceCont>{price}</PriceCont>
            </CollectionFooterCont>
            <CustomButtonCont inverted onClick={() => addItem(item)}>Add to Cart</CustomButtonCont>
        </CollectionItemCont>
    )
};

const mapDispatchToProps = dispatch => ({
    addItem: (item) => dispatch(addItem(item))
})

export default connect(null, mapDispatchToProps)(CollectionItem);