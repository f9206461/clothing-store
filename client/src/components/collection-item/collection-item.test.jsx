import React from "react";
import { shallow } from 'enzyme';
import { CollectionItem } from "./collection-item.component";

describe('collection-item', () => {
    const mockItem = {
        name: 'jacket', 
        price: 90, 
        imageUrl: 'https://www.testImage.com'
    };
    const addItem = jest.fn();
    const mockProps = { 
        item: mockItem, 
        addItem
    }
    const wrapper = shallow(<CollectionItem {...mockProps} />);

    it('should render CollectionItem component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should call addItem when CustomButton is clicked', () => {
        wrapper.find('CustomButtonCont').simulate('click');
        expect(addItem).toHaveBeenCalled();
    });

    it('should render imageUrl as a prop on BackgroundImage', () => {
        expect(wrapper.find('ImgCont').prop('imageUrl')).toBe(mockItem.imageUrl);
    });
    
    it('should render name prop in NameContainer', () => {
        expect(wrapper.find('NameCont').text()).toBe(mockItem.name);
    });
    
    it('should render price prop in PriceContainer', () => {
        const price = parseInt(wrapper.find('PriceCont').text());
        expect(price).toBe(mockItem.price);
    });
})