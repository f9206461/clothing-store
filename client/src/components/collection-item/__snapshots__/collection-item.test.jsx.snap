// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection-item should render CollectionItem component 1`] = `
<CollectionItemCont>
  <ImgCont
    className="image"
    imageUrl="https://www.testImage.com"
  />
  <CollectionFooterCont>
    <NameCont>
      jacket
    </NameCont>
    <PriceCont>
      90
    </PriceCont>
  </CollectionFooterCont>
  <CustomButtonCont
    inverted={true}
    onClick={[Function]}
  >
    Add to Cart
  </CustomButtonCont>
</CollectionItemCont>
`;
